### ISSUE TITLE
[Feature] Conditional selection (where) operation

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#38]
## Summary

Add a `where` operation that selects elements from two tensors based on a condition tensor (boolean mask).

## Motivation

The `where` operation is essential for conditional processing: masking invalid values, applying different operations based on conditions, implementing attention masks, and threshold-based value replacement.

## Reference Implementations

- **PyTorch**: `torch.where(condition, input, other)` — [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.where.html)
- **NumPy**: `numpy.where(condition, x, y)` — [NumPy docs](https://numpy.org/doc/stable/reference/generated/numpy.where.html)
- **ONNX**: `Where` operator (opset 9+) — [ONNX spec](https://onnx.ai/onnx/operators/onnx__Where.html)

## Verification

- ✅ `torch.where` exists — VERIFIED (selects from input or other based on condition)
- ✅ ONNX `Where` operator exists (opset 9+)
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for where, conditional)

## Proposed API

```dart
// Select from x where condition is true, y otherwise
final result = TensorOps.where(condition, x, y);

// As TransformOp (replace values below threshold with constant)
final op = WhereOp.threshold(threshold: 0.5, replaceValue: 0.0);
final result = op.apply(tensor);
```

## Implementation Notes

- `condition` tensor should contain `0.0`/`1.0` values (boolean encoding)
- Output: `result[i] = condition[i] != 0 ? x[i] : y[i]`
- Support tensor-tensor and tensor-scalar variants
- SIMD-acceleratable
- Add `OperationCapabilities`:
  - `onnxOpType: 'Where'`
  - `onnxOpsetVersion: 9`
  - `pytorchEquivalent: 'torch.where'`

## Dependencies

- Element-wise comparison operators (separate issue) — for generating condition tensors

## ROI Assessment

**High** — Enables conditional operations, attention masking, and threshold-based processing. Commonly used in post-processing pipelines.
